galsim/chromatic.py

```diff
@@ -1542,12 +1627,20 @@ def __init__(self, *args, **kwargs):
         for obj in self.objlist:
             self.wave_list = np.union1d(self.wave_list, obj.wave_list)
 
+    def __eq__(self, other):
+        return (isinstance(other, galsim.ChromaticSum) and
+                self.objlist == other.objlist and
+                self.gsparams == other.gsparams)
+
+    def __hash__(self):
+        return hash(("galsim.ChromaticSum", tuple(self.objlist), self.gsparams))
+
     def __repr__(self):
-        return 'galsim.ChromaticSum(%r, gsparams=%r)'%(self.objlist, self.gsparams)
+        return 'galsim.ChromaticSum(%r, gsparams=%r)' % (self.objlist, self.gsparams)
 
     def __str__(self):
-        str_list = [ str(obj) for obj in self.objlist ]
-        return 'galsim.ChromaticSum([%s])'%', '.join(str_list)
+        str_list = [str(obj) for obj in self.objlist]
+        return 'galsim.ChromaticSum([%s])' % ', '.join(str_list)
 
     def evaluateAtWavelength(self, wave):
```

Comment:
Here's a place (just below) where we're inconsistent with the spaces around the listcomp.